ext/standard/string.c

```diff
@@ -2301,6 +2301,90 @@ PHP_FUNCTION(substr)
 }
 /* }}} */
 
+/* {{{ proto boolean str_begins(string haystack, string needle)
+   Checks if haystack begins with needle */
+PHP_FUNCTION(str_begins) {
+    zend_string *haystack, *needle;
+    int i;
+
+    ZEND_PARSE_PARAMETERS_START(2, 2)
+        Z_PARAM_STR(haystack)
+        Z_PARAM_STR(needle)
+    ZEND_PARSE_PARAMETERS_END();
+
+    if (needle->len > haystack->len) {
+        RETURN_BOOL(0);
+    }
+
+    for (i = 0; i < needle->len; i++) 
```

Comment:
`RETURN_BOOL(memcmp(ZSTR_VAL(haystack), ZSTR_VAL(needle), ZSTR_LEN(needle)) == 0);`

Generally these loops should be replaced by `memcmp` calls.

Reply:
Really good call here! @nikic I replaced the loops for starts_with. The ends_with are a little trickier so I figure I'll wait to do that until the RFC is approved (if it is approved)!